lua/barbar/bbye.lua

```diff
@@ -169,24 +169,27 @@ local bbye = {}
 --- @return nil
 function bbye.delete(action, force, buffer, mods)
   local buffer_number = type(buffer) == 'string' and bufnr(buffer) or tonumber(buffer) or get_current_buf()
-
   if buffer_number < 0 then
-    err("E516: No buffers were deleted. No match for " .. buffer)
-    return
+    return err("E516: No buffers were deleted. No match for " .. buffer)
   end
 
-  local is_modified = buf_get_option(buffer_number, 'modified')
+  local has_confirm --- @type boolean
 
-  local has_confirm = get_option'confirm'
+  -- try arguments first
   if type(mods) == 'table' then
-    has_confirm = has_confirm or mods.confirm
+    has_confirm = mods.confirm
   elseif mods then
-    has_confirm = has_confirm or mods:match('conf') ~= nil
+    has_confirm = mods:match('conf') ~= nil
+  end
+
+  -- get the option if no arguments were passed in
+  if has_confirm == nil then
+    has_confirm = get_option('confirm')
```

Comment:
This is a perf change. Doing `get_option` is comparatively expensive to just looking at `mods` first, in case `:confirm` is there.